streamalert_cli/manage_lambda/package.py

```diff
@@ -27,298 +26,144 @@
+        'pymsteams==0.1.12',
+    }
+
+    PRECOMPILED_LIBS = {               # Default precompiled dependencies
+        'aliyun-python-sdk-actiontrail==2.0.0',
+        'boxsdk[jwt]==2.6.1',
     }
 
     def __init__(self, config):
         self.config = config
+        self.temp_package_path = os.path.join(tempfile.gettempdir(), self.package_name)
+
+    def _copy_user_config_files(self):
+        paths = set()
+        for location in self.CONFIG_EXTRAS:
+            paths.update(self.config['global']['general'].get(location, set()))
 
-    @property
-    def _required_libs(self):
-        return self.default_required_libs.union(self.package_libs)
+        self._copy_files(paths, ignores={'*.json'})
 
     def create(self):
         """Create a Lambda deployment package .zip file."""
         LOGGER.info('Creating package for %s', self.package_name)
 
-        temp_package_path = os.path.join(tempfile.gettempdir(), self.package_name)
-        if os.path.exists(temp_package_path):
-            shutil.rmtree(temp_package_path)
+        if os.path.exists(self.temp_package_path):
+            shutil.rmtree(self.temp_package_path)
 
-        self._copy_files(temp_package_path)
+        # Copy all of the default package files
+        self._copy_files(self.DEFAULT_PACKAGE_FILES, ignores={'*dependencies.zip'})
 
-        if not self._resolve_libraries(temp_package_path):
-            LOGGER.exception('Failed to install necessary libraries')
-            sys.exit(1)
+        # Copy in any user-specified files
+        self._copy_user_config_files()
+
+        if not self._resolve_libraries():
+            LOGGER.error('Failed to install necessary libraries')
+            return False
 
         # Extract any precompiled libs for this package
-        if self.precompiled_libs and not self._extract_precompiled_libs(temp_package_path):
-            LOGGER.exception('Failed to extract precompiled libraries')
-            sys.exit(1)
+        if not self._extract_precompiled_libs():
+            LOGGER.error('Failed to extract precompiled libraries')
+            return False
 
-        # Zip up files
-        # Build these in the top-level of the terraform directory
+        # Zip it all up
+        # Build these in the top-level of the terraform directory as streamalert.zip
         result = shutil.make_archive(
-            os.path.join(TERRAFORM_FILES_PATH, self.package_name), 'zip', temp_package_path)
-        LOGGER.info('Successfully created %s', os.path.basename(result))
+            os.path.join(TERRAFORM_FILES_PATH, self.package_name),
+            'zip',
+            self.temp_package_path
+        )
+
+        LOGGER.info('Successfully created package: %s', result)
 
         # Remove temp files
-        shutil.rmtree(temp_package_path)
+        shutil.rmtree(self.temp_package_path)
 
-        return True
+        return result
 
-    def _copy_files(self, temp_package_path):
+    def _copy_files(self, paths, ignores=None):
```

Comment:
Can you add the comment for arguments? It will be helpful to understand how to ignore file paths.